diff --git a/dist/esm/staticFunctionMiddleware.js b/dist/esm/staticFunctionMiddleware.js
index 3f9b7db31b55634034ac348ef8812b62aa0ff922..eeeaa54ab8814b64e00ae09ef3611a4479f06a2d 100644
--- a/dist/esm/staticFunctionMiddleware.js
+++ b/dist/esm/staticFunctionMiddleware.js
@@ -53,7 +53,7 @@ var fetchItem = async ({ data, functionId }) => {
 		functionId,
 		hash: jsonToFilenameSafeString(data)
 	});
-	return fetch(url, { method: "GET" }).then((r) => r.json()).then((d) => fromJSON(d, { plugins: getSerovalPlugins(defaultSerovalDeserializerPlugins) }));
+	return fetch(`${(process.env.TSS_ROUTER_BASEPATH ?? "").replace(/^\/*/, "/").replace(/\/+$/, "")}${url}`, { method: "GET" }).then((r) => r.json()).then((d) => fromJSON(d, { plugins: getSerovalPlugins(defaultSerovalDeserializerPlugins) }));
 };
 var staticFunctionMiddleware = createMiddleware({ type: "function" }).client(async (ctx) => {
 	if (process.env.NODE_ENV === "production" && typeof document !== "undefined") {
